boards/posix/native_posix/Kconfig.board

```diff
@@ -1,13 +1,24 @@
 # SPDX-License-Identifier: Apache-2.0
 
-config BOARD_NATIVE_POSIX
-	bool "Native POSIX"
+config BOARD_NATIVE_POSIX_32BIT
+	bool "Native POSIX for 32-bit host"
 	depends on SOC_POSIX
```

Comment:
As this is common for both, both it make sense to move it to BOARD_NATIVE_POSIX?